integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/AbstractBlazeJPAQuery.java

```diff
@@ -231,22 +231,35 @@ public String getQueryString() {
             }
         }
 
-        for (Map.Entry<String, Object> entry : hints.entries()) {
+        for (Map.Entry<String, Object> entry : getHintsInternal()) {
             criteriaBuilder.setProperty(entry.getKey(), entry.getValue().toString());
         }
 
-        if (cachable) {
+        if (cacheable) {
             criteriaBuilder.setCacheable(true);
         }
 
         return queryable;
     }
 
+    private Collection<Map.Entry<String, Object>> getHintsInternal() {
+        try {
+            Object hints = AbstractJPAQuery.class.getDeclaredField("hints").get(this);
```

Comment:
Can we cache this in a static initializer somehow? Also, would it be possible to make the guava dependency optional?

Reply:
Yes, I was thinking about that as well. Can we just store `AbstractJPAQuery.class.getDeclaredField("hints")` in a static variable? Or should we introduce a `HintsAccessor` class?

Reply:
Moving the logic to two dedicated classes allows to use a maven optional dependency on guava which is probably nicer, but I'll leave it up to you.

Reply:
As far as Guava: `blaze-persistence-integration-querydsl-expressions` never had an explicit dependency on Guava. So if QueryDSL is replaced with a version that does not ship Guava, you won't have it. I guess that will cause issues with importing it here as well. That can be easily avoided if we use reflection to call `entries()` as well.

Reply:
Right, but if we add an explicit optional dependency to the pom, this should work fine with both versions.

Reply:
Please let me now if I moved into the right direction 😄